cmake/Modules/Finded25519.cmake

```diff
@@ -0,0 +1,47 @@
+add_library(ed25519 UNKNOWN IMPORTED)
+
+find_path(ed25519_INCLUDE_DIR ed25519/ed25519.h)
+mark_as_advanced(ed25519_INCLUDE_DIR)
+
+find_library(ed25519_LIBRARY ed25519)
+mark_as_advanced(ed25519_LIBRARY)
+
+find_package_handle_standard_args(ed25519 DEFAULT_MSG
+    ed25519_INCLUDE_DIR
+    ed25519_LIBRARY
+    )
+
+set(URL https://github.com/warchant/ed25519.git)
+set(VERSION 0d31b082e0f0d4144fe598fb0f2a120e36de2a8f)
+set_target_description(ed25519 "Digital signature algorithm" ${URL} ${VERSION})
+
+if (NOT ed25519_FOUND)
+  externalproject_add(warchant_ed25519
+      GIT_REPOSITORY ${URL}
+      GIT_TAG        ${VERSION}
+      CMAKE_ARGS
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DEDIMPL=ref10
```

Comment:
It is always better to explicitly specify how exactly you use this library, I think. 
I explicitly say that I am using portable implementation and sha3.

Reply:
> Since commit is fixed, these values will not be changed without a manual update.
I agree with @lebdron. It is always time-consuming to explicitly specify, if the version does not change. Please, make it easier and follow advice, @Warchant